be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -805,6 +807,15 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
                 context->next_operator_id(), request.output_sink().mysql_table_sink,
                 mysql_table_sink->get_output_expr(), dop, fragment_ctx);
         fragment_ctx->pipelines().back()->add_op_factory(op);
+    } else if (typeid(*datasink) == typeid(starrocks::MemoryScratchSink)) {
+        auto* memory_scratch_sink = down_cast<starrocks::MemoryScratchSink*>(datasink.get());
```

Comment:
Can memory_scratch_sink be parallelized ? in non-pipeline engine, is this Sink one-instance?  maybe multiple memory_scratch_sinks can emit  RecordBatches on the same wire concurrently.

Reply:
> Can memory_scratch_sink be parallelized ? in non-pipeline engine, is this Sink one-instance? maybe multiple memory_scratch_sinks can emit RecordBatches on the same wire concurrently.

This sink can be parallelized, but it is determined by flink/spark. for example, if SR has multiple partitions and flink has multile source tasks, flink will decide which task is responsible for reading which data. The each source task will open a scanner to obtain data. for BE, each scanner is equivalent to a fragment to be executed. so we don't need to consider the parallelism problem.